Kernel/ACPI/Definitions.h

```diff
@@ -0,0 +1,262 @@
+    u32 oem_revision;
+    u32 creator_id;
+    u32 creator_revision;
+} __attribute__((__packed__));
+
+struct RSDT {
+    SDTHeader h;
+    u32 table_ptrs[1];
+} __attribute__((__packed__));
+
+struct XSDT {
+    SDTHeader h;
+    u64 table_ptrs[1];
+} __attribute__((__packed__));
+
+struct GenericAddressStructure {
+    u8 address_space;
+    u8 bit_width;
+    u8 bit_offset;
+    u8 access_size;
+    u64 address;
+} __attribute__((__packed__));
+
+struct FADT {
+    SDTHeader h;
+    u32 firmware_ctrl;
+    u32 dsdt_ptr;
+    u8 reserved;
+    u8 preferred_pm_profile;
+    u16 sci_int;
+    u32 smi_cmd;
+    u8 acpi_enable_value;
+    u8 acpi_disable_value;
+    u8 s4bios_req;
+    u8 pstate_cnt;
+    u32 PM1a_EVT_BLK;
+    u32 PM1b_EVT_BLK;
+    u32 PM1a_CNT_BLK;
+    u32 PM1b_CNT_BLK;
+    u32 PM2_CNT_BLK;
+    u32 PM_TMR_BLK;
+    u32 GPE0_BLK;
+    u32 GPE1_BLK;
+    u8 PM1_EVT_LEN;
+    u8 PM1_CNT_LEN;
+    u8 PM2_CNT_LEN;
+    u8 PM_TMR_LEN;
+    u8 GPE0_BLK_LEN;
+    u8 GPE1_BLK_LEN;
+    u8 GPE1_BASE;
+    u8 cst_cnt;
+    u16 P_LVL2_LAT;
+    u16 P_LVL3_LAT;
+    u16 flush_size;
+    u16 flush_stride;
+    u8 duty_offset;
+    u8 duty_width;
+    u8 day_alrm;
+    u8 mon_alrm;
+    u8 century;
+    u16 ia_pc_boot_arch_flags;
+    u8 reserved2;
+    u32 flags;
+    GenericAddressStructure reset_reg;
+    u8 reset_value;
+    u16 arm_boot_arch;
+    u8 fadt_minor_version;
+    u64 x_firmware_ctrl;
+    u64 x_dsdt;
+    GenericAddressStructure x_pm1a_evt_blk;
+    GenericAddressStructure x_pm1b_evt_blk;
+    GenericAddressStructure x_pm1a_cnt_blk;
+    GenericAddressStructure x_pm1b_cnt_blk;
+    GenericAddressStructure x_pm2_cnt_blk;
+    GenericAddressStructure x_pm_tmr_blk;
+    GenericAddressStructure x_gpe0_blk;
+    GenericAddressStructure x_gpe1_blk;
+    GenericAddressStructure sleep_control;
+    GenericAddressStructure sleep_status;
+    u64 hypervisor_vendor_identity;
+
+} __attribute__((__packed__));
+
+struct MADT : public SDTHeader {
+};
+
+struct DSDT : public SDTHeader {
+};
+
+struct PCI_MMIO_Descriptor {
+    u64 base_addr;
+    u16 seg_group_number;
+    u8 start_pci_bus;
+    u8 end_pci_bus;
+    u32 reserved;
+} __attribute__((__packed__));
+
+struct MCFG {
+    SDTHeader header;
+    u64 reserved;
+    PCI_MMIO_Descriptor descriptors[];
+} __attribute__((__packed__));
+}
+
+class ACPIStaticParser;
+
+namespace ACPI {
+
+class SDT : public RefCounted<SDT> {
+};
+
+struct GenericAddressStructure {
+    u8 address_space;
+    u8 bit_width;
+    u8 bit_offset;
+    u8 access_size;
+    u64 address;
+    GenericAddressStructure operator=(GenericAddressStructure other)
+    {
+        this->address_space = other.address_space;
+        this->bit_width = other.bit_width;
+        this->bit_offset = other.bit_offset;
+        this->access_size = other.access_size;
+        this->address = (u32)other.address;
+        return *this;
+    }
+    GenericAddressStructure operator=(ACPI_RAW::GenericAddressStructure other)
```

Comment:
This signature should be
```suggestion
    GenericAddressStructure& operator=(const ACPI_RAW::GenericAddressStructure& other)
```